src/iface/interface/mod.rs

```diff
@@ -1275,9 +1291,14 @@ impl<'a> Interface<'a> {
                     _ => unreachable!(),
                 }),
                 #[cfg(feature = "socket-udp")]
-                Socket::Udp(socket) => socket.dispatch(inner, |inner, response| {
-                    respond(inner, IpPacket::Udp(response))
-                }),
+                Socket::Udp(socket) => {
+                    socket.dispatch(inner, |inner, (rx_packet_id, ip_repr, udp_repr, data)| {
+                        respond(
```

Comment:
`respond` always makes up its own ID, AFAICT the `rx_packet_id` that gets stuffed into `IpPacket::Udp` is not used (?)

Reply:
Yep, not sure what happened here. Same as [above](https://github.com/smoltcp-rs/smoltcp/pull/628#discussion_r1059975699).